File: server/src/main/java/com/ahmadramadhan/mudahtitip/dto/SaleRequest.java
package com.ahmadramadhan.mudahtitip.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * DTO for recording a sale.
 */
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class SaleRequest {

    private Long consignmentId;
    private Integer quantity;
    private String notes;
}

File: client/lib/features/sale/data/models/sale_request.dart
import 'package:equatable/equatable.dart';

/// Request model for recording a sale.
class SaleRequest extends Equatable {
  final int consignmentId;
  final int quantity;
  final String? notes;

  const SaleRequest({
    required this.consignmentId,
    required this.quantity,
    this.notes,
  });

  Map<String, dynamic> toJson() {
    return {
      'consignmentId': consignmentId,
      'quantity': quantity,
      if (notes != null) 'notes': notes,
    };
  }

  @override
  List<Object?> get props => [consignmentId, quantity, notes];
}
